VTCareService/src/main/java/edu/vt/cs/vtcare/vtcareservice/dao/ProviderDao.java

```diff
@@ -38,4 +49,69 @@ public long persistProvider(Provider provider) throws SQLException {
             throw new SQLException("Encountered problem creating a new provider ", e);
         }
     }
+
+    /**
+     *
+     * @param id of provider we're looking for
+     * @return list of providers with only 1 entry, when providerID is known
+     * @throws SQLException on SQL errors like table not existing
+     */
+    public List<Provider> getProvider(long id) throws SQLException {
+        try (PreparedStatement statement = connection.prepareStatement(FIND_BY_PROVIDER_ID_SQL)) {
+            statement.setLong(1, id);
+
+            ResultSet res = statement.executeQuery();
+            List<Provider> items = new ArrayList<>();
+
+            if(res.next() ){
+                extractProvider(res, items);
+            }
+            return items;
+        } catch (SQLException e) {
+            throw new SQLException("Encountered problem fetching a provider ", e);
+        }
+    }
+
+    /**
+     *
+     * @return ListOfAllProviders
+     * @throws SQLException When query is malformed, or other SQL related issues.
+     */
+    public List<Provider> getProviders() throws SQLException{
+        try (PreparedStatement statement = connection.prepareStatement(FIND_ALL_PROVIDERS_SQL)) {
+
+            ResultSet res = statement.executeQuery();
+            List<Provider> items = new ArrayList<>();
+
+            while(res.next() ){
+                extractProvider(res, items);
+            }
+            return items;
+        } catch (SQLException e) {
+            throw new SQLException("Encountered problem fetching providers ", e);
+        }
+    }
+
+    /**
+     *
+     * @param res resultset to iterate through, used to construct provider
+     * @param items data structure used to contain providers
+     * @throws SQLException
+     */
+    private void extractProvider(ResultSet res, List<Provider> items) throws SQLException {
+        Provider prov = new Provider(res.getObject("name", String.class) ,
+                res.getObject("email", String.class) ,
```

Comment:
The ResultSet class have an elaborate API to fetch appropriate kind of values from it. So for example you are trying to fetch a string type here so use res.getString("email") and like wise for other types like getInt() etc. You can check all the methods with the . operator. 
```suggestion
                res.getObject("email", String.class) ,
```

Reply:
Yeah that's my mistake, I saw the first method for getString to column id, and didn't want to hardcode it in. I just saw there's also one that takes column names.

Reply:
I have already added a method to extract provider - which is now available in main branch. So, you can use that instead as is.